src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,44 @@
+package duke;
+
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+/**
+ * Handles the interaction the user and the program.
+ */
+class Ui {
+    private Scanner scanner;
+    Ui() {
+        scanner = new Scanner(System.in);
+    }
+
+    protected String getLine() throws NoSuchElementException {
```

Comment:
I think getNextLine or even just nextLine would be a better name, getLine implies getting a specific line from a line number